core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CipherHolder.java

```diff
@@ -24,9 +24,14 @@
 /**
  * Holds ciphertext with related metadata.
  *
+ * @deprecated This is deprecated since version 4.6.1.
+ * This is replaced with org.wso2.carbon.crypto.api.CipherMetaDataHolder.
+ * This is deprecated due to moving self contained cipher text creation logic to crypto-service project.
+ *
  * IMPORTANT: this is replicated at org.wso2.carbon.user.core.config.UserStoreConfigXMLProcessor.CipherHolder,
- *              what ever changes applied here need to update on above. This is done to avoid cyclic dependency.
+ * what ever changes applied here need to update on above. This is done to avoid cyclic dependency.
```

Comment:
I guess this will not apply as we will now have it in crypto service layer.
Still keep a note and verify

Reply:
Since the class deprecated, I think this will not matter. Yes, there is a cyclic dependecy between carbon.core and carbon.user.core. Therefore in user.core we have to add a new CipherMetaDataHolder instead of this deprecated one.